app/models/assignment.rb

```diff
@@ -20,6 +20,40 @@ class Assignment < ActiveRecord::Base
 
   delegate :name, to: :split, prefix: true
 
+  class << self
+    def to_hash
+      Hash[all.includes(:split).map { |a| [a.split.name.to_sym, a.variant.to_sym] }]
+    end
+
+    def for_presentation(built_at: nil)
+      q = presentation_query
+
+      q = built_at.nil? ? q.where("splits.finished_at is null") : q.where("splits.finished_at > ?", built_at)
+
+      q
```

Comment:
Any reason not to make this a `scope`? I'm also wondering if it is worth trying to deconstruct into subparts (especially in anticipating additional variants) or if it is better to leave it in one big colocated query method the way you have it:

```ruby
scope :for_presentation, ->(built_at: nil) { undecided.unfinished(as_of: built_at) }

# assignment.rb
scope :undecided, -> { joins(:split).where('splits.decided_at IS NULL OR assignments.updated_at > splits.decided_at') }
scope :unfinished(as_of:) -> { joins(:split).merge(Split.unfinished(as_of: as_of)) }

# split.rb
scope :unfinished, ->(as_of:) { as_of ? where('splits.finished_at > ?', as_of) : where(finished_at: nil) }
```

(reactions to that?)

Reply:
yeah, this looks good. I'm inclined to walk toward this in subsequent PRs, if that makes sense, 'cuz I've already got another working PR based on this factoring. I like the composable scopes thing, effectively they're not great developer experience on their own but as long as I expose macro scopes that do everything in a coordinated way that's fine.

Stylistically, I generally have preferred class methods whenever a scope-like thing takes args because the syntax gets a little nuts. I think one of the rails core folks has a blog post about how that's his preference as well. Do you have a reason you prefer scopes even with args?

Reply:
Also, if I keep the composable-and-yet-not-ideal-developer-experience scopes as private class methods, that's probably a win too.

Reply:
My sole reason for preferring `scope` is that it tells me exactly what it returns, so I don't have to read the guts and do the variable-hopping to determine that oh, it in fact returns an ActiveRecord relation and is basically a `scope` in sheep's clothing.

Reply:
But I totally buy that single-use monster scopes work better on net as class methods. We can defer this to later PRs when we really know what we're dealing with here.

Reply:
Also, private scopes should be a thing.